runtime/lua/vim/lsp.lua

```diff
@@ -607,6 +608,12 @@ do
         })
       end
     end
+    vim.schedule(function()
```

Comment:
If this isn't wrapped in a deferred callback, then any `LspDidChange` autocmds aren't executed when changes are undone in normal mode.